plugins/RobotController/RobotController.cpp

```diff
@@ -38,6 +38,16 @@ RobotController::RobotController()
     , pImpl{new Impl(), [](Impl* impl) { delete impl; }}
 {}
 
+RobotController::~RobotController()
+{
+    auto* ecSingleton = EnvironmentCallbacksSingleton::Instance();
+
+    if (bool removed = ecSingleton->deleteEnvironmentCallback(pImpl->robot->name()); !removed) {
```

Comment:
There is any reason for not keeping `bool removed = ecSingleton->deleteEnvironmentCallback(pImpl->robot->name())` outside the `if` loop? In this way the code is a bit heard to read (to be honest I am just assuming what `if(a;b)` actually means, I do not know without checking.

Reply:
This is new C++17 syntax. Since we typically use this style of checking errors in an `if` statement and then omit the `else` for the code after the check, I prefer this syntax particularly when I collapse the code in the IDE. There's no external resource that leaks.

Just to understand, you don't like this `if` style, or you just are not used to it and it seems strange?

Reply:
> Just to understand, you don't like this if style, or you just are not used to it and it seems strange?

Definitely the second one, I  am just not used to it. Apparently it is also quite difficult to find info on that was also tricky to find resources on it, unless you know that this is C++17. On my machine, if I search on Google "semicolon inside if clause C++" beside cppreference I don't get any relevant result (I found something more relevant if I add C++17). However, it is just that developers need to get used to this, not an actual problem of the PR.